dbms/src/TestUtils/tests/gtest_mock_executors.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/FmtUtils.h>
+#include <Flash/Coprocessor/DAGQuerySource.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/executeQuery.h>
+#include <TestUtils/InterpreterTestUtils.h>
+#include <TestUtils/mockExecutor.h>
+
+namespace DB
+{
+namespace tests
+{
+class MockDAGRequestTest : public DB::tests::MockExecutorTest
+{
+};
+
+TEST_F(MockDAGRequestTest, MockTable)
+try
+{
+    auto builder = DAGRequestBuilderFactory().createDAGRequestBuilder();
```

Comment:
How about migrating `mock table` to `DAGRequestBuilderFactory`?
such as 
```
auto builder_factory = DAGRequestBuilderFactory();
build.add_mock_table({"test_db", "test_table"}, {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}});

auto left_builder = build_factory.createDAGRequestBuilder();
left_builder.mock_table("test_db", "test_table");

auto right_builder = build_factory.createDAGRequestBuilder();
right_builder.mock_table("test_db", "test_table").join(left_builder);
```

We can avoid defining `test_db.test_table` twice

Reply:
Good idea! It will be helpful to maintain all metadata in the context.

Reply:
Maybe  the following interface is more readable.
```
auto builder_factory = DAGRequestBuilderFactory();
build.add_mock_table({"test_db", "test_table"}, {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}});

auto left_builder = build_factory.createDAGRequestBuilder();
left_builder.use("test_db").table("test_table");

auto right_builder = build_factory.createDAGRequestBuilder();
right_builder.use("test_db").table("test_table").join(left_builder);
```